### PR TITLE
Dasd 8218 add premium func app plan template

### PR DESCRIPTION
**Pull request by NickGraham101**

### CODE REVIEW

File: templates/premium-plan.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        "EP3"
+      ],
+      "defaultValue": "EP1"
+    }
+  },
+  "variables": {},
+  "resources": [
+    {
+      "type": "Microsoft.Web/serverfarms",
+      "apiVersion": "2016-09-01",
```
Comment (nbowes24) [inline templates/premium-plan.json:34]: Should we try use the latest for new templates? 2020-06-01. Started using it recently for resource group deployments.

Comment (NickGraham101) [reply templates/premium-plan.json]: Upgraded to latest API version

File: templates/function-app.json [hunk 2]
```diff
@@ -64,7 +64,7 @@
         "serverFarmId": "[variables('appServicePlanId')]",
         "clientAffinityEnabled": false,
         "siteConfig": {
-          "alwaysOn": "[if(equals(reference(resourceId(parameters('appServicePlanResourceGroup'),'Microsoft.Web/serverFarms',parameters('appServicePlanName')), '2019-08-01').computeMode,'Dynamic'), 'false', 'true')]",
+          "alwaysOn": "[if(contains(createArray('functionapp','elastic'),reference(resourceId(parameters('appServicePlanResourceGroup'),'Microsoft.Web/serverFarms',parameters('appServicePlanName')), '2019-08-01').kind), 'false', 'true')]",
```
Comment (jack-education) [inline templates/function-app.json:67]:
```suggestion
          "alwaysOn": "[if(equals('elastic',reference(resourceId(parameters('appServicePlanResourceGroup'),'Microsoft.Web/serverFarms',parameters('appServicePlanName')), '2019-08-01').kind), 'false', 'true')]",
```

Comment (NickGraham101) [reply templates/function-app.json:67]: That'll break Consumption plans.  Neither Consumption nor Premium plans can accept a function app with `alwaysOn` set to `true` so we need to set to `false` for apps that are deploying to an ASP that has the value of their `Kind` property set to either `functionapp` (a Consumption plan) or `elastic` (a Premium plan)

**Review by jack-education — APPROVED** [review]

**Pull request merged**